File: 백준/Silver/19583. 싸이버개강총회/싸이버개강총회.py
import sys
input = sys.stdin.readline

def time_compare(t1, t2):
    m1 = int(t1[:2]) * 60 + int(t1[3:])
    m2 = int(t2[:2]) * 60 + int(t2[3:])
    return True if m1 >= m2 else False

def solution():

    S, E, Q = input().strip().split()
    check = {}

    while (log := sys.stdin.readline()):
        time, name = log.strip().split()

        if time_compare(S, time):
            check[name] = 0

        else:
            if time_compare(time, E) and time_compare(Q, time):
                if name in check:
                    check[name] += 1

    answer = 0
    for val in check.values():
        if val:
            answer += 1
    print(answer)


solution()

File: 백준/Silver/19583. 싸이버개강총회/싸이버개강총회.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        String S = st.nextToken();
        String E = st.nextToken();
        String Q = st.nextToken();
        String log = "";

        String time = "";
        String name = "";

        Set<String> check = new HashSet<>();
        int answer = 0;

        while ((log = br.readLine()) != null) {
            String[] strarr = log.split(" ");
            time = strarr[0];
            name = strarr[1];

            if (timeCompare(S, time)) {
                check.add(name);
            } else if (timeCompare(time, E) && timeCompare(Q, time)) {
                if (check.contains(name)) {
                    answer++;
                    check.remove(name);
                }
            }
        }
        System.out.println(answer);
        
    }

    public static boolean timeCompare(String time1, String time2) {
        int m1 = Integer.parseInt(time1.substring(0, 2)) * 60 + Integer.parseInt(time1.substring(3));
        int m2 = Integer.parseInt(time2.substring(0, 2)) * 60 + Integer.parseInt(time2.substring(3));
        return m1 >= m2;
    }
}